junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/NodeTestTask.java

```diff
@@ -165,13 +165,25 @@ private void cleanUp() {
 
 	private void reportCompletion() {
 		if (throwableCollector.isEmpty() && skipResult.isSkipped()) {
+			try {
+				node.nodeSkipped(context, testDescriptor, skipResult);
+			}
+			catch(Throwable t) {
+				//swallow
+			}
 			listener.executionSkipped(testDescriptor, skipResult.getReason().orElse("<unknown>"));
 			return;
 		}
 		if (!started) {
 			// Call executionStarted first to comply with the contract of EngineExecutionListener.
 			listener.executionStarted(testDescriptor);
 		}
+		try {
+			node.nodeFinished(context, testDescriptor, throwableCollector.toTestExecutionResult());
+		}
+		catch(Throwable t) {
+			//swallow
```

Comment:
If `nodeFinished()` throws an exception, let's at least log it a `DEBUG` level.

Reply:
Would you suggest that its an additional logging (as there is already in the method implementation), or should i remove that one? Since Node is a general abstraction but testwatchers are limited to jupiter extension api, i will need to know when the error comes from a watcher. Maybe a new exception type?

Reply:
Although it's overkill for the specific case of `TestWatcher` with the `JupiterTestEngine`, I think we should still log any exceptions thrown here for the sake of use with other third-party engine implementations.

@junit-team/junit-lambda, do you agree?